util/validateparam.go

```diff
@@ -0,0 +1,23 @@
+package util
+
+import (
+	"fmt"
+	"log"
+)
+
+func ValidateStringParam(paramName string, paramValue string, validValues []string) {
+	validValueMap := make(map[string]bool)
+	for _, validValue := range validValues {
+		validValueMap[validValue] = true
+	}
```

Comment:
This still has the overhead of `create array in caller -> create map from that array here -> check value`. If you create the maps in the caller directly, you won't need to do the extra step of conversion.

Reply:
True, but I think I would prefer to leave it as-is in this case since it's really a negligible optimization for a list that might contain maybe 5 params, tops. I would rather provide an easier interface for passing in valid values.

Reply:
That's fair :)